dvc/remote/local/__init__.py

```diff
@@ -569,3 +569,48 @@ def _get_unpacked_dir_names(self, checksums):
             if self.is_dir_checksum(c):
                 unpacked.add(c + self.UNPACKED_DIR_SUFFIX)
         return unpacked
+
+    def _needs_checkout(self, path_info, checksum, force):
+        if self._can_avoid_copy(path_info, checksum):
```

Comment:
also it is not clear why are you wrapping _can_avoid_copy into _needs_checkout. Feels like it is unncecessary.

Reply:
As discussed privately: 
"can avoid copy" indicates cache link type which is local remote domain, hence wrapping.

Reply:
remeved that method